src/main_benchmark.cpp

```diff
@@ -0,0 +1,411 @@
+//*****************************************************************************
+// Copyright 2022 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <algorithm>
+#include <chrono>
+#include <future>
+#include <iomanip>
+#include <iostream>
+#include <numeric>
+#include <sstream>
+#include <thread>
+#include <vector>
+
+#include <cxxopts.hpp>
+#include <signal.h>
+#include <stdio.h>
+#include <sysexits.h>
+
+#include "pocapi.h"  // NOLINT
+#include "stringutils.hpp"
+
+namespace {
+
+class BenchmarkCLIParser {
```

Comment:
Why to wrap it to `BenchmarkCLIParser` if we can use `cxxopts::ParseResult` directly?